cmd/argocd/commands/app.go

```diff
@@ -1037,50 +1037,14 @@ func NewApplicationWaitCommand(clientOpts *argocdclient.ClientOptions) *cobra.Co
 
 // printAppResources prints the resources of an application in a tabwriter table
 // Optionally prints the message from the operation state
-func printAppResources(w io.Writer, app *argoappv1.Application, showOperation bool) {
-	messages := make(map[string]string)
-	opState := app.Status.OperationState
-	var syncRes *argoappv1.SyncOperationResult
-
-	if showOperation {
-		fmt.Fprintf(w, "GROUP\tKIND\tNAMESPACE\tNAME\tSTATUS\tHEALTH\tHOOK\tMESSAGE\n")
-		if opState != nil {
-			if opState.SyncResult != nil {
-				syncRes = opState.SyncResult
-			}
-		}
-		if syncRes != nil {
-			for _, res := range syncRes.Resources {
-				if !res.IsHook() {
-					messages[fmt.Sprintf("%s/%s/%s/%s", res.Group, res.Kind, res.Namespace, res.Name)] = res.Message
-				} else if res.HookType == argoappv1.HookTypePreSync {
-					fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", res.Group, res.Kind, res.Namespace, res.Name, res.HookPhase, "", res.HookType, res.Message)
-				}
-			}
-		}
-	} else {
-		fmt.Fprintf(w, "GROUP\tKIND\tNAMESPACE\tNAME\tSTATUS\tHEALTH\n")
-	}
+func printAppResources(w io.Writer, app *argoappv1.Application) {
```

Comment:
Looks like CLI won't print sync message any more. Can we restore this functionality?

Reply:
let me check

Reply:
This was a tough one. A resource can be a hook, and a hook may have two messages (in edge cases, they can be part of two phases). 

Options:

* No message.
* Only messages for non-hook resources.
* Combine multiple messages into a summary message.

Thoughts?

Reply:
I think we need to combine messages into a summary message.